### PR TITLE
docs: Added CHANGELOG & code of conduct before first release candidate.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Contributor Covenant Code of Conduct
```
Comment (brancz) [inline CODE_OF_CONDUCT.md]: :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Rule (including alerts) evaluation logic for Ruler (experimental).
+- StoreAPI logic for Ruler.
+- Basic metric orchestration for all components.
+- Verify commands with potential fixes (experimental).
+- Compact / Downsample offline commands.
+- Bucket commands.
+- E2E tests for query part.
+- Getting start, design & components documentation.
+- Workflow tools (makefile) and CI file for Circle CI.
+
```
Comment (domgreen) [inline CHANGELOG.md]: add at the last line?
```
[Unreleased]: https://github.com/improbable-eng/thanos/compare/v0.1.0-rc.0...HEAD
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- Downsampling logic for Compactor (experimental).
+- Rule (including alerts) evaluation logic for Ruler (experimental).
+- StoreAPI logic for Ruler.
+- Basic metric orchestration for all components.
+- Verify commands with potential fixes (experimental).
+- Compact / Downsample offline commands.
+- Bucket commands.
+- E2E tests for query part.
+- Getting start, design & components documentation.
+- Workflow tools (makefile) and CI file for Circle CI.
```
Comment (brancz) [inline CHANGELOG.md]: not relevant to users who want to use a release of Thanos

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- Support for S3 minio-based AWS object storage (experimental).
+- Compaction logic of blocks from multiple sources for Compactor.
+- Downsampling logic for Compactor (experimental).
+- Rule (including alerts) evaluation logic for Ruler (experimental).
+- StoreAPI logic for Ruler.
+- Basic metric orchestration for all components.
+- Verify commands with potential fixes (experimental).
+- Compact / Downsample offline commands.
+- Bucket commands.
+- E2E tests for query part.
```
Comment (brancz) [inline CHANGELOG.md]: not relevant to users who want to use a release of Thanos

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- Optimized object storage fetch logic for Store.
+- Index cache and chunk pool for Store for better memory usage.
+- Stable support for Google Cloud Storage object storage.
+- StoreAPI logic for Querier to support Thanos federation (experimental).
+- Support for S3 minio-based AWS object storage (experimental).
+- Compaction logic of blocks from multiple sources for Compactor.
+- Downsampling logic for Compactor (experimental).
+- Rule (including alerts) evaluation logic for Ruler (experimental).
+- StoreAPI logic for Ruler.
+- Basic metric orchestration for all components.
```
Comment (brancz) [inline CHANGELOG.md]: what do you mean by this, can you elaborate?

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+- Verify commands with potential fixes (experimental).
+- Compact / Downsample offline commands.
+- Bucket commands.
+- E2E tests for query part.
+- Getting start, design & components documentation.
```
Comment (brancz) [inline CHANGELOG.md]: I'd handle documentation out of band. People are not interested in changes in docs of a patch release. They should just (eventually) be available for each minor version release in rendered form and only the latest of each minor release.

File: CHANGELOG.md (hunk 5 above)
Comment (Bplotka) [reply CHANGELOG.md]: just that all the components should have some metrics exposed that allows to measure the perf/success rate and alerting (:

File: CHANGELOG.md (hunk 2 above)
Comment (Bplotka) [reply CHANGELOG.md]: Maybe in the first line?

**Pull request merged**